src/Tmds.Ssh/KerberosCredential.cs

```diff
@@ -0,0 +1,294 @@
+    private extern static void GetMICMethod(NegotiateAuthentication context, ReadOnlySpan<byte> data, IBufferWriter<byte> writer);
+#endif
+
+    /// <summary>
+    /// Create a Kerberos credential using gssapi-with-mic.
+    /// </summary>
+    /// <remarks>
+    /// The credential specified is used for the Kerberos authentication process. This can either be the same or
+    /// different from the username specified through <c>SshClientSettings.UserName</c>. The client settings username
+    /// is the target login user the SSH service is meant to run as, whereas the credential here is the Kerberos
+    /// principal used for authentication. The rules for how a Kerberos principal maps to the target user is defined by
+    /// the SSH service itself. For example on Windows the username should be the same but on Linux the mapping can be
+    /// done through a <c>.k5login</c> file in the target user's home directory.
+    ///
+    /// If the credential is null, the username in the credential is null/empty, or the password is null/empty, the
+    /// Kerberos authentication will be done using the cached credential. For Windows this means the current thread's
+    /// identity (typically logon user) will be used. For *nix this will use the Kerberos CCACHE principal that can
+    /// be managed using the <c>kinit</c> command. If there is no available cache credential, the authentication will
+    /// fail.
+    ///
+    /// Credentials can only be delegated if the Kerberos ticket retrieved from the KDC is marked as forwardable.
+    /// Windows hosts will always retrieve a forwardable ticket but non-Windows hosts may not. When using an explicit
+    /// credential, make sure that 'forwardable = true' is set in the krb5.conf file so that .NET will request a
+    /// forwardable ticket required for delegation. When using a cached credential, make sure that when the ticket was
+    /// retrieved it was retrieved with the forwardable flag. If the ticket is not forwardable, the authentication will
+    /// still work but the ticket will not be delegated.
+    /// </remarks>
+    /// <param name="credential">The credentials to use for the Kerberos authentication exchange. Set to null to use the cached credential.</param>
+    /// <param name="delegateCredential">Request delegation on the Kerberos context.</param>
+    /// <param name="serviceName">Override the service principal name (SPN), default uses the <c>host/{connection.HostName}.</c></param>
+    public KerberosCredential(NetworkCredential? credential = null, bool delegateCredential = false, string? serviceName = null)
+    {
+        _kerberosCredential = credential ?? CredentialCache.DefaultNetworkCredentials;
+        _delegateCredential = delegateCredential;
+        _serviceName = serviceName;
+    }
+
+    internal async Task<bool> TryAuthenticate(SshConnection connection, ILogger logger, SshClientSettings settings, SshConnectionInfo connectionInfo, CancellationToken ct)
```

Comment:
When required libraries are missing (`krb5`) this method shouldn't throw and return `false`.